pkg/ctl/enable/repo.go

```diff
@@ -22,84 +18,45 @@ func enableRepo(cmd *cmdutils.Cmd) {
 		"Set up a repo for gitops, installing Flux in the cluster and initializing its manifests in the specified Git repository",
 		"",
 	)
-	var opts flux.InstallOpts
+	opts := ConfigureRepositoryCmd(cmd)
 	cmd.SetRunFuncWithNameArg(func() error {
-		if err := opts.GitOptions.ValidateURL(); err != nil {
-			return errors.Wrap(err, "please supply a valid --git-url argument")
-		}
-		if opts.GitOptions.Email == "" {
-			return errors.New("please supply a valid --git-email argument")
-		}
-		if opts.GitPrivateSSHKeyPath != "" && !file.Exists(opts.GitPrivateSSHKeyPath) {
-			return errors.New("please supply a valid --git-private-ssh-key-path argument")
-		}
-
-		if err := cmdutils.NewInstallFluxLoader(cmd).Load(); err != nil {
-			return err
-		}
-		cfg := cmd.ClusterConfig
-		ctl, err := cmd.NewCtl()
-		if err != nil {
-			return err
-		}
-
-		if err := ctl.CheckAuth(); err != nil {
-			return err
-		}
-		if ok, err := ctl.CanOperate(cfg); !ok {
-			return err
-		}
-		kubernetesClientConfigs, err := ctl.NewClient(cfg)
-		if err != nil {
-			return err
-		}
-		k8sConfig := kubernetesClientConfigs.Config
-
-		k8sRestConfig, err := clientcmd.NewDefaultClientConfig(*k8sConfig, &clientcmd.ConfigOverrides{}).ClientConfig()
-		if err != nil {
-			return errors.Wrap(err, "cannot create Kubernetes client configuration")
-		}
-		k8sClientSet, err := kubeclient.NewForConfig(k8sRestConfig)
-		if err != nil {
-			return errors.Errorf("cannot create Kubernetes client set: %s", err)
-		}
-
-		installer := flux.NewInstaller(k8sRestConfig, k8sClientSet, &opts)
-		userInstructions, err := installer.Run(context.Background())
-		logger.Info(userInstructions)
-		return err
+		return Repository(cmd, opts)
```

Comment:
I think by convention we name these "doEnableRepo()" but I don't have a very strong opinion about it

Reply:
I was initially planning to have the logic from `create/cluster.go` call these methods, hence it would read `enable.Repository` and `enable.Profile`. However, I'm currently unsure what is best between doing that or use the previous logic from `enableProfile` (which has now moved to `all.go`):

1. calling `enable.Repository` and `enable.Profile` has the benefits of being _exactly_ what it would be to call these two commands, but doesn't leverage `gitops.Applier` ;
2. using the previous `enableProfile` logic / the current logic from `all.go` keeps the higher-level abstraction that we created for that purpose: `gitops.Applier`.

That being said, if we don't want to anticipate too much `create cluster -f` in this PR, I can revert to the naming we used to have. This is, again, because this work was almost done/in progress, before we decided to abort the API changes. Let me know if you have a strong opinion about this, and I can clean-up more "eagerly".

Reply:
I would avoid calling these methods from another file in the same "layer". Every time I did that in the past ended up biting me in the ass because I was creating a tight couple between two things that didn't need it. So even if it may look a bit repetitive I would have `create cluster -f` do its own thing without depending on this file. 

One potential example is the loader: `create cluster -f` does its own loading of the config file while `enable.Repository` has its own loader.

Reply:
FWIW, I made them private again.